routes/candidateRoutes.js

```diff
@@ -121,13 +136,30 @@ candidateRouter.put('/update/id=:id', authorizedAdmin, async (req, res) => {
 
 candidateRouter.delete('/delete/id=:id', authorizedAdmin, async (req, res) => {
     const candidateId = req.params.id;
-    Candidate.deleteOne({ id: Number(candidateId) }).then(() => {
+    //Getting the image url from db before deleting it
+    const findImgUrl = await Candidate.findOne({ id: Number(candidateId) });
+    if(findImgUrl){
+        const imgRegex = /[^/]+$/;
+        const imgUrl = RegExp(imgRegex).exec(findImgUrl.img);
+        const params = {
+            Bucket: BUCKET_NAME,
+            Key: imgUrl[0],
+        };
+        const command = new DeleteObjectCommand(params);
+        const imgDeleted = await s3.send(command);
+        if(!imgDeleted){
+            logger.error("Unable to delete image from s3");
+            return res.status(500).send({error: 'Unable to delete image'});
```

Comment:
Perhaps 424 is even better for all the Amazon failing cases. Check if it's an official status

Reply:
Dont think 424 is an official status code for amazon

Reply:
I think 500 is fine for this case